Signal/src/call/CallAudioService.swift

```diff
@@ -238,16 +250,16 @@ import AVFoundation
         stopAnyRingingVibration()
     }
 
-    private func play(sound: Sound, call: SignalCall) {
+    private func play(sound: Sound) {
         guard let newPlayer = sound.player else {
             Logger.error("\(self.TAG) unable to build player")
+            assertionFailure()
             return
         }
         Logger.info("\(self.TAG) playing sound: \(sound.filePath)")
 
-        newPlayer.play()
         self.currentPlayer?.stop()
-
+        newPlayer.play()
```

Comment:
This re-ordering is apparently critical. I'm actually not sure why.

I could imagine why it would fail if we were calling play(sound: Sound.incomingRing) multiple times - but we aren't. 

Without it, on iOS9, with the app in the foreground, the ringer would only be audible 2 in 5 times. With this re-ordering I heard proper audio 5 in 5 times.